lock.h

```diff
@@ -10,11 +10,11 @@
 #define BOOL _WINBOOL
 #	include <windows.h>
 #undef BOOL
-typedef HANDLE mutex_t;
-#	define INIT_LOCK(x) x = CreateMutex(NULL, FALSE, NULL)
-#	define LOCK(x) WaitForSingleObject(*x, INFINITE)
-#	define UNLOCK(x) ReleaseMutex(*x)
-#	define DESTROY_LOCK(x) CloseHandle(*x)
+typedef CRITICAL_SECTION mutex_t;
+#	define INIT_LOCK(x) InitializeCriticalSection(&(x))
+#	define LOCK(x) EnterCriticalSection(x)
```

Comment:
critical sections can time out after about 10 minutes. this was deemed an acceptable cost.

Reply:
No objection to this, but I wondered if you'd done any profiling of Mutex vs CriticalSection vs SRW lock for workloads that you care about.

Reply:
The data there is, unfortunately, lost to the ages. @bviglietta made the switch for us ~2 years ago.